domain/modelagent/service/service.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/version/v2"
+
+	"github.com/juju/juju/core/model"
+)
+
+// State provides the state methods needed by the modelagent service.
+type State interface {
+	GetModelAgentVersion(ctx context.Context, modelID model.UUID) (version.Number, error)
+}
+
+// Service is a modelagent service which can be used to get the running Juju
+// agent version for any given model.
+type Service struct {
+	st State
+}
+
+// NewService returns a new modelagent service.
+func NewService(st State) *Service {
+	return &Service{st: st}
+}
+
+// GetModelAgentVersion returns the agent version for the specified model. If
+// the agent version cannot be found, an error satisfying [errors.NotFound]
+// will be returned.
+func (s *Service) GetModelAgentVersion(ctx context.Context, modelID model.UUID) (version.Number, error) {
+	if err := modelID.Validate(); err != nil {
+		return version.Zero, errors.Annotate(err, "validating model UUID")
```

Comment:
Can we please change UUID to ID? One of the jobs coming up is to rename this type. Just trying to make less work with UUID removal.